src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/PseHelper.java

```diff
@@ -150,4 +178,68 @@ private static String formatClaimantReply(PseResponseType pseResponseType,
+            return "";
+        }
+
+        IntWrapper respondCount = new IntWrapper(0);
+        return detailsWrapper("Tribunal Responses", sendNotificationType.getRespondNotificationTypeCollection().stream()
+                .filter(r -> filterTribunalResponse(r.getValue(), party))
+                .map(r -> formatTribunalReply(r.getValue(), respondCount))
+                .collect(Collectors.joining("")));
+    }
+
+    private static boolean filterTribunalResponse(RespondNotificationType value, String party) {
+        return switch (party) {
+            case CLAIMANT_TITLE ->
+                    value.getRespondNotificationPartyToNotify().equals(CLAIMANT_ONLY)
+                    || value.getRespondNotificationPartyToNotify().equals(BOTH_PARTIES)
+                    || value.getRespondNotificationWhoRespond().equals(CLAIMANT_ONLY)
+                    || value.getRespondNotificationWhoRespond().equals(BOTH_PARTIES);
+            case RESPONDENT_TITLE ->
+                    value.getRespondNotificationPartyToNotify().equals(RESPONDENT_ONLY)
+                    || value.getRespondNotificationPartyToNotify().equals(BOTH_PARTIES)
+                    || value.getRespondNotificationWhoRespond().equals(RESPONDENT_ONLY)
+                    || value.getRespondNotificationWhoRespond().equals(BOTH_PARTIES);
+            default -> throw new IllegalArgumentException(INVALID_PARTY_SELECTION);
+        };
+    }
+
+    private static String formatTribunalReply(RespondNotificationType value, IntWrapper respondCount) {
+        return "\r\n" + MarkdownHelper.createTwoColumnTable(
+                new String[]{"Tribunal Response " + respondCount.incrementAndReturnValue(), " "},
+                Stream.of(
+                        asRow("Notification", defaultIfEmpty(value.getRespondNotificationTitle(), " - ")),
+                        asRow("Response from", defaultIfEmpty(value.getRespondNotificationFullName(), " - ")),
+                        asRow("Response date", defaultIfEmpty(value.getRespondNotificationDate(), " - ")),
+                        addDocumentTypeRows(value.getRespondNotificationUploadDocument(), "Supporting material"),
+                        asRow("Additional information",
+                                defaultIfEmpty(value.getRespondNotificationAdditionalInfo(), " - ")),
+                        asRow("Response Type", defaultIfEmpty(value.getRespondNotificationCmoOrRequest(), " - ")),
+                        asRow("Party to notify", defaultIfEmpty(value.getRespondNotificationPartyToNotify(), " - ")),
+                        asRow("Is a response required?",
+                                defaultIfEmpty(value.getRespondNotificationResponseRequired(), " - ")),
+                        asRow("Parties to respond", defaultIfEmpty(value.getRespondNotificationWhoRespond(), " - "))
+                ));
+    }
+
+    public static boolean getPartyNotifications(SendNotificationTypeItem sendNotificationTypeItem, String party) {
```

Comment:
How about renaming this method to something that suits a boolean return type, probably after splitting it to something like "isClaimantPartyNotification" or "isRespondentPartyNotification"?

Reply:
- Updated method name to _canPartyViewNotification_
- Not splitting the method into a claimant and respondent version as it will cause a lot of duplicated code around populating dynamic lists and filtering the collection which is why I used the approach of passing through the type of party as a parameter

Reply:
accepted